core/cli/commands/structural_fusion.py

```
@@ -0,0 +1,400 @@
+    auth = base64.b64encode(f"{username}:{password}".encode()).decode()
+    base_url = f"http://{host}:{port}/_db/{database}"
+
+    payload: dict[str, Any] = {"query": query}
+    if bind_vars:
+        payload["bindVars"] = bind_vars
+    data = json.dumps(payload).encode()
+    req = urllib.request.Request(
+        f"{base_url}/_api/cursor",
+        data=data,
+        headers={"Authorization": f"Basic {auth}", "Content-Type": "application/json"},
+    )
```

Ruff (0.15.4)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)